x/stakeibc/keeper/icqcallbacks_withdrawal_balance_test.go

```diff
@@ -0,0 +1,264 @@
+		EpochNumber:        1,
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 30_000_000_000), // dictates timeouts
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), strideEpochTracker)
+
+	withdrawalBalance := int64(1000)
+	commission := uint64(10)
+	expectedReinvestment := sdk.NewCoin(Atom, sdk.NewInt(int64(900)))
+
+	params := s.App.StakeibcKeeper.GetParams(s.Ctx())
+	params.StrideCommission = uint64(commission)
+
+	queryRequest := s.CreateBalanceQueryRequest(withdrawalAddress, Atom)
+	queryResponse := s.CreateBalanceQueryResponse(withdrawalBalance, Atom)
+
+	return WithdrawalBalanceICQCallbackTestCase{
+		initialState: WithdrawalBalanceICQCallbackState{
+			hostZone: hostZone,
+			withdrawalChannel: Channel{
+				PortID:    icatypes.PortPrefix + withdrawalAccountOwner,
+				ChannelID: withdrawalChannelId,
+			},
+			withdrawalBalance: withdrawalBalance,
+		},
+		validArgs: WithdrawalBalanceICQCallbackArgs{
+			query: icqtypes.Query{
+				Id:      "0",
+				ChainId: HostChainId,
+				Request: queryRequest,
+			},
+			callbackArgs: queryResponse,
+		},
+		expectedReinvestment: expectedReinvestment,
+	}
+}
+
+func (s *KeeperTestSuite) TestWithdrawalBalanceCallback_Successful() {
+	tc := s.SetupWithdrawalBalanceCallbackTest()
+
+	// Get the sequence number before the ICA is submitted to confirm it incremented
+	withdrawalChannel := tc.initialState.withdrawalChannel
+	withdrawalPortId := withdrawalChannel.PortID
+	withdrawalChannelId := withdrawalChannel.ChannelID
+
+	startSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx(), withdrawalPortId, withdrawalChannelId)
+	s.Require().True(found, "sequence number not found before reinvestment")
+
+	// Call the ICQ callback
+	err := stakeibckeeper.WithdrawalBalanceCallback(s.App.StakeibcKeeper, s.Ctx(), tc.validArgs.callbackArgs, tc.validArgs.query)
+	s.Require().NoError(err)
+
+	// Confirm ICA reinvestment callback data was stored
+	s.Require().Len(s.App.IcacallbacksKeeper.GetAllCallbackData(s.Ctx()), 1, "number of callbacks found")
+	callbackKey := icacallbackstypes.PacketID(withdrawalPortId, withdrawalChannelId, startSequence)
+	callbackData, found := s.App.IcacallbacksKeeper.GetCallbackData(s.Ctx(), callbackKey)
+	s.Require().True(found, "callback data was not found for callback key (%s)", callbackKey)
+	s.Require().Equal("reinvest", callbackData.CallbackId, "callback ID")
+
+	// Confirm reinvestment callback args
+	callbackArgs, err := s.App.StakeibcKeeper.UnmarshalReinvestCallbackArgs(s.Ctx(), callbackData.CallbackArgs)
+	s.Require().NoError(err, "unmarshalling callback args error for callback key (%s)", callbackKey)
+	s.Require().Equal(tc.initialState.hostZone.ChainId, callbackArgs.HostZoneId, "host zone in callback args (%s)", callbackKey)
+	s.Require().Equal(tc.expectedReinvestment, callbackArgs.ReinvestAmount, "reinvestment coin in callback args (%s)", callbackKey)
```

Comment:
thoughts on checking that the ICA sequence number has incremented?

Reply:
ah yes, I meant to add that! done